internal/comm/catchup_handler_test.go

```diff
@@ -247,3 +247,110 @@ func TestCatchupHandler_ServeHTTP_Blocks(t *testing.T) {
 		require.Equal(t, uint64(6), bNum)
 	})
 }
+
+func TestCatchupHandler_ServeHTTP_LargeResponse(t *testing.T) {
+	lg, err := logger.New(&logger.Config{
+		Level:         "debug",
+		OutputPath:    []string{"stdout"},
+		ErrOutputPath: []string{"stderr"},
+		Encoding:      "console",
+	})
+	require.NoError(t, err)
+
+	b1Size := 0
+	b5Size := 0
+
+	ledger1 := &memLedger{}
+	for n := uint64(1); n < 11; n++ {
+		block := &types.Block{
+			Header: &types.BlockHeader{
+				BaseHeader:           &types.BlockHeaderBase{Number: n},
+				TxMerkelTreeRootHash: make([]byte, 1024), //just for size
+			},
+		}
+		ledger1.Append(block)
+
+		if n==1 {
+			b1Size = len(httputils.MarshalOrPanic(block))
+		}
+		if n<6 {
+			b5Size += len(httputils.MarshalOrPanic(block))
+		}
+	}
+
+	t.Run("too many blocks in request", func(t *testing.T) {
+		h := comm.NewCatchupHandler(lg, ledger1, b5Size+1) // 5 blocks in response
+		require.NotNil(t, h)
+
+		resp := httptest.NewRecorder()
+		req, _ := http.NewRequest(http.MethodGet, comm.GetBlocksPath, nil)
+		q := req.URL.Query()
+		q.Add("start", "2")
+		q.Add("end", "9") // too many
+		req.URL.RawQuery = q.Encode()
+		req.Header.Set("Accept", httputils.MultiPartFormData)
+
+		h.ServeHTTP(resp, req)
+		require.Equal(t, http.StatusOK, resp.Result().StatusCode)
+
+		_, params, err := mime.ParseMediaType(resp.Result().Header.Get("Content-Type"))
+		require.NoError(t, err)
+		boundary, ok := params["boundary"]
+		require.True(t, ok)
+		mr := multipart.NewReader(resp.Result().Body, boundary)
+		bNum := uint64(2)
+		for part, errP := mr.NextPart(); errP == nil; part, errP = mr.NextPart() {
```

Comment:
we can have `for { if errP == io.EOF {break}}` for better readability. We also need to ensure that no err other than `io.EOF` is returned. The same is applicable for the other sub-test with a smaller maxSize.